### PR TITLE
refactor provider settings to get data from credentials endpoint

### PR DESCRIPTION
**Pull request by anotherchrisberry**
Still need to determine what's required on the installation script side of things (if anything). As written, everything should should "work" without changing the scripts:
 * if there are no configured preferred zones in cloud driver, all zones will be used, which is a sensible default
 * primary accounts won't exist (unless that work has made it into the scripts), so all account tags would appear gray.

@zanthrash please review

### CODE REVIEW

**Comment by anotherchrisberry** [comment]
@cfieber Is this mergeable or are there ramifications for non-Netflix users?

File: app/scripts/modules/core/account/account.service.js [hunk 1]
```diff
@@ -11,38 +11,37 @@ module.exports = angular.module('spinnaker.core.account.service', [
...
+              if (providerName === 'azure') {
+                preferredZones = [region.name];
+              }
+              if (region.preferredZones) {
+                preferredZones = _.intersection(region.preferredZones, region.availabilityZones);
+              }
+              preferredZonesByAccount[account.name][region.name] = preferredZones;
+            });
+          });
+          return preferredZonesByAccount;
```
Comment (zanthrash) [inline app/scripts/modules/core/account/account.service.js]: Usually like to see a 'catch' block on promises. Either here or on the "getAllAccountDetailsForProvider" call.

**Comment by zanthrash** [comment]
Looks good.

File: app/scripts/modules/core/account/account.service.js (hunk 1 above)
Comment (anotherchrisberry) [inline app/scripts/modules/core/account/account.service.js]: Done (added to getAllAccountDetailsForProvider method, with tests)

**Pull request merged**